code-challenges.test.js

```diff
@@ -19,33 +27,239 @@ const people = [
 
 // a) Create a test with an expect statement using the variables provided.
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+describe("cubeSummer", () => {
 
+  const cubeAndSum1 = [2, 3, 4];
+  const cubeAndSum2 = [0, 5, 10];
+
+  it("takes in an array of numbers and returns the sum of all the numbers cubed", () => {
+    expect(cubeSumer(cubeAndSum1)).toEqual(99);
+    expect(cubeSumer(cubeAndSum2)).toEqual(1125);
+  });
+});
+
+// ReferenceError: cubeSummer is not defined; good fail :)
 
 // b) Create the function that makes the test pass.
+
+// PSEUDOCODE:
+// NOTE: the problem is asking to cube each number FIRST, then add them all up
+  // EXAMPLE: (2 * 2 * 2) + (3 * 3 * 3) + (4 * 4 * 4) = 99
+// create function called cubeSumer
+  // takes in an array
+// create a variable called sum 
+// iterate through array with .map()
+  // cube each value with i ** 3
+    // array.map((value) => value ** 3)
+// add all values with array.reduce((previousValue, currentValue) => previousValue + currentValue, sum variable)
+  // .reduce() passes the return value to the next index   
+// return sum
+
+const cubeAndSum1 = [2, 3, 4];
+const cubeAndSum2 = [0, 5, 10];
+
+// FIRST ATTEMPT
+
+// const cubeSumer = (array) => {
+//   let sum = array.map((value) => value ** 3).reduce((previousValue, currentValue) => previousValue + currentValue); // <----- multiple steps here, in one line; first, I use .map() to cube each value with arrow function. Then I use .reduce() to add all the values together. The syntax for .reduce() is: "array.reduce((previousValue, currentValue) => previousValue + currentValue)" . This adds the whole array together and returns a single number.
+//   console.log(sum) // <----- console.loggin so I can see the magic
+//   return sum;
+// }
+
+
+// SECOND ATTEMPT
+// This is the same code as my first attempt, but it shows how you can do the same thing on multiple lines. I think it's more readable this way. Both work just fine, though.
+
+const cubeSumer = (array) => {
+  let sum = array.map((value) => value ** 3) // <---- cube each value with i ** 3
+  let sum2 = sum.reduce((previousValue, currentValue) => previousValue + currentValue); // <--- add all values with .reduce()
+  // console.log(sum2) // <----- console.loggin 
+  return sum2;
+}
+
+console.log(cubeSumer(cubeAndSum1));
+console.log(cubeSumer(cubeAndSum2));
+
+// RESULT:
+// console.log
+
+// 99
+// at cubeSumer (code-challenges.test.js:145:11)
+
+// 1125
+// at cubeSumer (code-challenges.test.js:145:11)
+
+// What I learned: You can chain multiple arrow functions together. Originally, I planned to separate the .map() and .reduce() into separate functions. Also, console.log() is amazing because it allows you to see what is happening and where. I also learned to STOP using return prematurely. That has been tripping me up for a while now.
```

Comment:
Great work. Please place whether or not your functions passed

Reply:
I mentioned it on Line 214

Reply:
I see that good failure for the test which is needed. We also expect to see when the function passes, which occurs after you create it.